### PR TITLE
Return fire tweaks ported from BN

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
Bugfixes "Return fire tweaks ported from BN."

#### Purpose of change
- Reuse gun actor instead of copypasting it.
- Fix several bugs.

#### Describe the solution
Cherry-picked changes from https://github.com/cataclysmbnteam/Cataclysm-BN/pull/545.

#### Describe alternatives you've considered
None.

#### Testing
Debug-spawned character with night vision, debug-spawned M9, debug-spawned CROWS M249. Fired at it in full darkness. Got return fire.

#### Additional context
None.

### CODE REVIEW

File: src/mondefense.cpp [hunk 1]
```diff
@@ -164,44 +164,36 @@ void mdefense::return_fire( monster &m, Creature *source, const dealt_projectile
...
-    npc tmp;
-    tmp.set_fake( true );
-    tmp.setpos( m.pos() );
+    const int distance_to_source = rl_dist( m.pos(), source->pos() );
 
-    // We might be aiming at the player square, but we aren't totally sure where they are,
-    // so represent that with initial recoil.
-    tmp.recoil = 150;
+    // TODO: implement different rule, dependent on sound and probably some other things
+    // Add some innacuracy since it is blind fire
```
Comment (actual-nh) [inline src/mondefense.cpp:181]:
```suggestion
    // Add some inaccuracy since it is blind fire (at a tile, not the player directly)
```

**Review by actual-nh — COMMENTED** [review]
Any interactions with the new body parts code (no limbs = major penalty)?

**Comment by Night-Pryanik** [comment]
> Any interactions with the new body parts code (no limbs = major penalty)?

I'm not yet familiar with the new code, maybe in the future.

**Comment by actual-nh** [comment]
> > Any interactions with the new body parts code (no limbs = major penalty)?
> 
> I'm not yet familiar with the new code, maybe in the future.

I was asking because there was a bug with the fake NPCs for turrets of lacking limbs, thus turrets started missing a lot. On another look - if you're using the same actor as for regular fire, that should do the job. Kevin was finding earlier some problems with the turret return-fire code.

**Pull request merged**